src/main/scala/co/ledger/wallet/daemon/libledger_core/database/JDBCResultSet.scala

```diff
@@ -0,0 +1,176 @@
+package co.ledger.wallet.daemon.libledger_core.database
+
+import java.sql.{PreparedStatement, ResultSet, Types}
+
+import co.ledger.core._
+
+/**
+  *
+  *
+  * User: Pierre Pollastri
+  * Date: 17-01-2019
+  * Time: 09:33
+  *
+  */
+
+sealed trait RichDatabaseResultSet extends DatabaseResultSet {
+  override def getRow: DatabaseResultRow = null
+  override def getUpdateCount: Int = 0
+  override def available(): Int = 0
+  override def hasNext: Boolean = false
+  override def next(): Unit = throw new RuntimeException("Trying to iterate rows from an empty database result set")
+  override def close(): Unit = () // No op
+  // scalastyle:off
+  override def getError: DatabaseError = null
+  def describeColumn(i: Int): DatabaseColumn = {
+    if (i >= 0) null else null
+  }
+  // scalastyle:on
+  def getColumnCount: Int = 0
+}
+
+class JDBCResultSet(statement: PreparedStatement, queryType: JDBCQueryType) extends RichDatabaseResultSet {
+  override def getRow: DatabaseResultRow = _set.getRow
+  override def getUpdateCount: Int = _set.getUpdateCount
+  override def available(): Int = _set.available()
+  override def hasNext: Boolean = _set.hasNext
+  override def next(): Unit = _set.next()
+  override def close(): Unit = _set.close()
+  override def getError: DatabaseError = _set.getError
+  override def describeColumn(i: Int): DatabaseColumn = _set.describeColumn(i)
+  override def getColumnCount: Int = _set.getColumnCount
+
+  private val _set: UnifiedResultSet = {
```

Comment:
Maybe it's safer to make it `lazy`?

Reply:
I don't understand what do you want to put lazy ?

Reply:
the statement is executed as long as the object is initiated. I don't know if that's an intended behaviour

Reply:
Yep that's intended

Reply:
I would have this `JDBCResultSet` as a summoner of `UnifiedResultRet` or directly `DatabaseResultSet`.
Something like this:
```suggestion
object DatabaseResultSet {
    def execute(statement: PreparedStatement, queryType: JDBCQueryType): DatabaseResultSet = {
        try {
       statement.execute()
       queryType match {
         case JDBCSelectQueryType =>
           val rs = statement.getResultSet
           val isEmpty = !rs.isBeforeFirst
           QueryResultSet(rs, isEmpty, new JDBCDatabaseResultRow(rs, 1))
         case _ =>
           UpdateResultSet(statement.getUpdateCount)
       }
     } catch {
       case all: Throwable =>
         all.printStackTrace()
         throw all
     }
    }
}
```

Reply:
I don't get what you want to do